client/core/trade.go

```diff
@@ -2854,8 +2854,14 @@ func (t *trackedTrade) confirmRedemption(match *matchTracker) {
 		}
 		return
 	}
-	if !t.wallets.toWallet.connected() {
-		return
+
+	// In some cases the wallet will need to send a new redeem transaction.
+	didUnlock, err := t.wallets.toWallet.refreshUnlock()
+	if err != nil { // Just log it and try anyway.
+		t.dc.log.Errorf("refreshUnlock error checking redeem %s: %v", t.wallets.toWallet.Symbol, err)
+	}
+	if didUnlock {
+		t.dc.log.Warnf("Unexpected unlock needed for the %s wallet to check a redemption", t.wallets.toWallet.Symbol)
 	}
```

Comment:
Since this may need to create a new redemption, for eth too, I think we need this check.

Reply:
Adding. For the peerCount, the xcWallet has this field but should it be checked live or good enough to use the field?

Reply:
Current field value is good.